# 2021-11-30

- Modified the renaming scripts to function better;
- Created a python script to order the sequences in the alignments to perform the protein-guided alignments (had to rename first);
- Generated protein-guided alignments using the existing protein files and aligning with MAFFT:
```bash
for i in *proteins*; do mafft --auto $i > ${i%.fas}_aln.fas; done
for i in *proteins_sorted_aln*; do pal2nal $i ${i%%_*}_cds* -output fasta > ${i%%_*}_cds_sorted_aln.fas; done
```
- Aligned with MAFFT:
	- 45168at33392 - ok;
	- 75072at33392 - trimmed the 3' end (there was a non-coding gap there in multiple sequences);
	- ~~67937at33392~~ - internal stop codon at 5';
	- 65545at33392 - trimmed 5' end;
	- 73136at33392 - ok;
	- 87096at33392 - trimmed 5' and a big chunk of 3';
	- 75566at33392 - trimmed 5';
	- 89570at33392 - trimmed the first codon at 5';
	- 118220at33392 - trimmed the first codon at 5';
	- 116590at33392 - ok;
	- 37660at33392 - trimmed the 5' and 3' ends;
	- 105333at33392 - trimmed 5' and a big chunk of 3';
	- 87633at33392 - trimmed a large portion of 5', added a single nucleotide of missing data at 5' for GCF_019457755, trimmed the 3' end;
	- 68602at33392 - trimmed 5' and a big chunk of 3';
	- 61549at33392 - trimmed large portions of the 5' and 3' ends;
	- 74309at33392 - trimmed a large portion of 5', trimmed 3';
	- 87827at33392 - trimmed the 5' and 3' ends;
	- 95275at33392 - trimmed the 5' and 3' ends;
	- 24031at33392 - trimmed a large portion of 5', trimmed 3';
	- 112028at33392 - ok;
	- 61826at33392 - trimmed the 5' and 3' ends;
	- 79473at33392 - trimmed 5' end;
	- 47016at33392 - trimmed 5' and 3' ends;
	- 81580at33392 - trimmed 3' end;
	- ~~33501at33392~~ - doesn't exist anymore;
	- ~~59339at33392~~ - internal region with low-quality alignment, removed;
	- 56574at33392 - trimmed 5' end and a large portion of the 3' end;
	- 86363at33392 - trimmed first codon at 5' and a big chunk of the 3' end;
	- 61724at33392 - trimmed a large portion of the 5' end, trimmed the 3' end, and there's a large indel at near 3' in GCF_012274295 and GCF_004153925;
	- 100935at33392 - trimmed the 5' end and the last codon at 3';
	- 60560at33392 - trimmed big chunks of the 5' and 3' ends;
	- 114758at33392 - trimmed a decent chunk at 5';
	- 116109at33392 - trimmed 5' and 3' ends, alignment at 5' is not great and has many gaps, but they're all multiple of 3 and translation is fine;
	- 56271at33392 - trimmed a large portion of 5', trimmed 3';
	- 123921at33392 - trimmed a decent portion of the 5' end;
	- 100122at33392 - trimmed the 5' end;
	- 107482at33392 - trimmed a large portion of 5', trimmed 3';
	- 14326at33392 - trimmed a large chunk of the 5', alignment near 3' is not great and has many gaps, but they're all multiple of 3 and translation is fine;
	- ~~122669at33392~~ - internal stop codons and missing data, removed;
	- 43062at33392 - trimmed a large portion of the 5' end, and trimmed the 3' end;
	- 43658at33392 - trimmed a large portion of the 5' end, and trimmed the 3' end;
	- 124625at33392 - trimmed 5' and 3' ends;
	- 122095at33392 - trimmed a decent chunk of 5' and the last codon at 3';
- Generated the matrix files with SequenceMatrix;
- Generated the PartitionFinder config file for the full dataset;
- Created the two reduced datasets with codon positions removed;
- Generated the PartitionFinder inputs for the reduced datasets;
- Sent the files to ZÃ© and uploaded them to LGM;
- Ran PartitionFinder for the full dataset first:
```bash
nice -n 10 python ~/eusociality/bin/partitionfinder-2.1.1/PartitionFinder.py -p 50 partition_finder_full_dataset/
```