# 2021-04-27

- All of the following steps were executed inside LGM:
    - Ran a BUSCO analysis agains the model files:
		```bash
		nice -n 10 python /usr/local/src/busco/scripts/run_BUSCO.py -i ~/eusociality/results/maker_results/2021-04-26/GCF_003254395.2_Amel_HAv3.1_genomic.all.maker.model_gff%3Amaker.proteins.renamed.fasta -c 50 -o Amel_maker_test -m prot -l ~/eusociality/data/busco_data/2019-06-10/hymenoptera_odb9/ > busco_log.txt
		```
	- Ran a BLAST query against the *Apis mellifera* non-redundant proteome:
		```bash
		nice -n 10 blastp -query ../2021-04-26/GCF_003254395.2_Amel_HAv3.1_genomic.all.maker.model_gff%3Amaker.proteins.renamed.fasta -db ~/eusociality/data/blast_dbs/GCF_003254395.2_Amel_HAv3.1_genomic.gbff_nt.fasta.longest.fasta.ids.txt.aa.fa -outfmt 6 -out Amel_maker_model_vs_proteome.blastp -num_threads 50 -evalue 1e-6
		```

- Locally:
	- Polished and likely finished `preprocess_sra_mitogenome_data.sh`. Only need to integrate the genome assembler now;
	- Deleted some previous `samtools` results and reran some of the tools:
		- Recreated the SAM version of the file from the BAM file to check how to filter unaligned sequences;
		- Reran `samtools sort` withe the `-n` flag to sort by read name, so that the final fastq file is created correctly when using `samtools fastq`;
		- Actually, ended up messing a lot of things so re-ran the entire pipeline.