# 2021-04-18

- All of the following steps were executed inside LGM:
    - Followed the SNAP training steps up to the creation of the `export` files to train AUGUSTUS;
        - Created an `augustus` folder to store the files;
        - Ran `/usr/local/src/maker/src/bin/gff3_merge -d ../GCF_003254395.2_Amel_HAv3.1_genomic_master_datastore_index.log -o amel_genome.all.gff` to generate the GFF3 input files;
        - Ran `/usr/local/src/maker/src/bin/maker2zff amel_genome.all.gff`  to generate the `genome.ann` and `genome.dna` files;
        - Ran `fathom genome.ann genome.dna -validate > snap_validate_output.txt` to validate the gene models. Checked the validation results with `cat snap_validate_output.txt | grep "error"`. There were two erroneous models. Removed them with: `grep -vwE "MODEL10516" genome.ann > genome.ann2` and `grep -vwE "MODEL2203" genome.ann2 > genome.ann3`. Removed `genome.ann` and `genome.ann2` and renamed `genome.ann3` to `genome.ann`. Confirmed there were no errors with `fathom genome.ann2 genome.dna -validate`;
        - Generated the SNAP input files with the following sequence of commands:
            - `fathom genome.ann genome.dna -categorize 1000`;
            - `fathom uni.ann uni.dna -export 1000 -plus`;
        - Downloaded the script [**zff2augustus**](https://github.com/hyphaltip/genome-scripts/blob/master/gene_prediction/zff2augustus_gbk.pl);
        - Ran `zff2augustus_gbk.pl > augustus.gbk` inside the `augustus` directory;
        - Split train and test sets with `perl /usr/local/bin/augustus-3.1/scripts/randomSplit.pl augustus.gbk 100`;
        - Copied the AUGUSTUS config path to `augustus` with `cp -r /usr/local/bin/augustus-3.1/config/ .` and gave it permissions. Exported the new config directory with `export AUGUSTUS_CONFIG_PATH=./config/`;
        - Created the species config directory with `perl /usr/local/bin/augustus-3.1/scripts/new_species.pl --species=apis_mellifera`;
        - Trained AUGUSTUS with:
            - `etraining --species=apis_mellifera augustus.gbk.train`;
            - `augustus --species=apis_mellifera augustus.gbk.test | tee first_training.out`;
        - Optimized the model with `perl /usr/local/bin/augustus-3.1/scripts/optimize_augustus.pl --species=apis_mellifera augustus.gbk.train`.