# 2021-04-06

- Generated the config files for MAKER (in LGM);
    - Waiting for Chico to install and configure some of the dependencies;
- Created the input files for MAKER:
    - A set of `ESTS_CDS` files, in two versions, `with_amel` and `without_amel`, which are files containing concatenated EST and CDS sequences for all organisms. The two version respectively don't have sequence data for *Apis mellifera* but instead for another bee (*Megalopta genalis*); and sequences including *Apis mellifera*, but excluding *Megalopta genalis*. This is needed because the first annotation will be a benchmark and quality assessment on the *Apis mellifera* genome, so we need a non-biased dataset for that;
    - A set of `protein` files, in the same two versions as above, for the same reasons;
    - A set of `ESTS` files, in the same two versions as above, for the same reasons. This file only contains the EST contigs for the species, without the CDS sequences. This is just to serve as something like a backup;
- Grabbed the genomes for the species we are going to annotate (for now, only *Apis mellifera*);
- Fixed the paths to MAKER's dependencies and started running MAKER on the *Apis mellifera* genome;
    - Command: `nice -n 10 maker -c 50 -fix_nucleotides`;
    - Actually killed the process: should do the CD-HIT analysis first;
    - When running, should actually do: `nice -n 10 time maker -c 50 -fix_nucleotides`;
- Ran CDHIT on the file `ESTS_CDS_without_amel`;
    - Command: `~/eusociality/bin/cdhit/cd-hit-est -i ESTS_CDS_without_amel.fasta -o ESTS_CDS_without_amel_cdhit -c 1.0 -n 11 -d 0 -M 16000 -T 50`;
    - Results were underwhelming (about 1% of size reduction) so ran with 95% similarity: `nice -n 10 ~/eusociality/bin/cdhit/cd-hit-est -i ESTS_CDS_without_amel.fasta -o ESTS_CDS_without_amel_cdhit_95.fasta -c 0.95 -n 10 -d 0 -M 16000 -T 50`;
    - Not a big reduction again (about 4%), so we discarded the nucleotide results for CD-HIT;
- Ran CDHIT on the file `proteins_without_amel.fasta`:
    - Command: `nice -n 10 ~/eusociality/bin/cdhit/cd-hit -i proteins_without_amel.fasta -o proteins_without_amel_cdhit_100.fasta -c 1.0 -n 5 -d 0 -M 16000 -T 50`;
    - Again, not a big reduction. Tried with 95%: `nice -n 10 ~/eusociality/bin/cdhit/cd-hit -i proteins_without_amel.fasta -o proteins_without_amel_cdhit_95.fasta -c 0.95 -n 5 -d 0 -M 16000 -T 50`;
    - Not a big reduction, so dumped the CD-HIT results for proteins as well;
- Ran MAKER again, ignoring the CD-HIT tests;
    - Ran with: `nice -n 10 time maker -c 50 -fix_nucleotides`;
- Started working on a Python script to download mitochondrial genomes.