# 2021-10-13

- While reviewing the BUSCO results for the paper, discovered the the bee *Habropoda laboriosa* actually passes our cutoffs, so started working on it;
- Ran BUSCO for all species that were overlooked, first using the Hymenoptera dataset, then just for *H. laboriosa* using the Endopterygota dataset:
```bash
conda activate BUSCO5
cd /home/maycon/eusociality/results/busco_results/2021-11-29

# Create a folder to store all proteomes
mkdir tmp

nice -n 10 busco -i tmp/ -o incomplete_hymenoptera -m prot -l hymenoptera_odb10 -c 50
nice -n 10 busco -i tmp/ -o h_laboriosa -m prot -l endopterygota_odb10 -c 50
```
- Updated the metadata table with the new information and the BUSCO results;
- Started working on getting the BUSCO orthologs for the tree construction:
```bash
cd /home/maycon/eusociality/work/busco_phylo/
cp 2021-10-13 2021-11-29
cd 2021-11-29/tmp

# copy the genome that's missing
cp ~/eusociality/data/hymenoptera/2019-11-07/proteins/filtered_out/hymenoptera_filtered_out_proteins.tar.gz .
cp ~/eusociality/data/hymenoptera/2019-11-01/longest/hymenoptera_longest_original.tar.gz .
tar -xvzf hymenoptera_longest_original.tar.gz GCF_001263275.1_ASM126327v1_genomic.gbff_nt.fasta.longest.fasta
mv GCF_001263275.1_ASM126327v1_genomic.gbff_nt.fasta.longest.fasta cds
tar -xvzf hymenoptera_filtered_out_proteins.tar.gz GCF_001263275.1_ASM126327v1_genomic.gbff_nt.fasta.longest.fasta.ids.txt.aa.fa
mv GCF_001263275.1_ASM126327v1_genomic.gbff_nt.fasta.longest.fasta.ids.txt.aa.fa prot
rm hymenoptera_*

# get the ids for the BUSCO orthologs classified as "complete"
for file in $(find ~/eusociality/results/busco_results/2021-11-29/h_laboriosa/GCF_001263275.1_ASM126327v1_genomic.gbff_nt.fasta.longest.fasta.ids.txt.aa.fa/run_endopterygota_odb10/ -name "full_table*.tsv"); do grep -v "^#" ${file} | awk '$2=="Complete" {print $1}' >> complete_busco_ids.txt; done # H. laboriosa

# get the counts for each unique BUSCO id
sort complete_busco_ids.txt | uniq -c > complete_busco_ids_with_counts.txt

# filter all ids with a count < 54
# means we only keep those which are complete in all species, Hymenoptera + Coleoptera
awk '$1 > 53 && $1 < 55 {print $2}' complete_busco_ids_with_counts.txt > final_busco_ids_complete_in_54_with_outgroup.txt

# generate the input file for the python script
# getting the busco id, protein id and genome file name
for file in $(find ~/eusociality/results/busco_results/2021-10-13/endopterygota_52hymenoptera_1coleoptera_busco/*/run* -name "full_table*.tsv"); do short=$(echo "${file#*endopterygota_52hymenoptera_1coleoptera_busco/}"); echo "${short%%\.*}" >> protein_ids_complete_in_53_with_outgroup.txt; for id in $(cat final_busco_ids_complete_in_53_with_outgroup.txt); do grep -v "^#" ${file} | awk -v id="$id" '{OFS="|"} {if ($1==id) print $1,$3}' >> protein_ids_complete_in_53_with_outgroup.txt; done; done # all 53 species, including outgroup
for file in $(find ~/eusociality/results/busco_results/2021-11-29/h_laboriosa/*/run* -name "full_table*.tsv"); do short=$(echo "${file#*h_laboriosa/}"); echo "${short%%\.*}" >> protein_ids_complete_in_54_with_outgroup.txt; for id in $(cat final_busco_ids_complete_in_54_with_outgroup.txt); do grep -v "^#" ${file} | awk -v id="$id" '{OFS="|"} {if ($1==id) print $1,$3}' >> protein_ids_complete_in_54_with_outgroup.txt; done; done # H. laboriosa

# ran the python script to generate separate files for each ortholog
# complete in all species, containing the sequences for all 54 species
# for proteins and CDSs
python3.6 ~/eusociality/bin/utils/get_busco_genes.py protein_ids_complete_in_54_with_outgroup.txt tmp/cds/ tmp/prot/ genes/

# checked that all generated files have 54 sequences each
grep -c ">" genes/* | awk -F":" '{OFS=":"} {if ($2 != 54) print $1,$2}'
```
- In total, 503 complete orthologs were found;
- Ran alignments for all CDSs using MAFFT with the `--auto` option:
```bash
mkdir alignments
nice -n 10 time ~/eusociality/bin/utils/./run_alignments.sh genes/ alignments/ 50
```
- Updated the names to id file to add the new species;
- Ran the script to fetch the OrthoDB data for the BUSCO orthologs to download the data for the 503 genes complete in all 54 species:
```bash
# in /mnt/2A2A43C42A438C2F/lab_data/busco_phylo/2021-11-29
scp -P 2772 maycon@150.164.28.21:/home/maycon/eusociality/work/busco_phylo/2021-10-13/final_busco_ids_complete_in_53_with_outgroup.txt .

# in /home/maycon/Documents/LAB/eusociality/local_work
python get_ortodb_data.py /mnt/2A2A43C42A438C2F/lab_data/busco_phylo/2021-11-29/final_busco_ids_complete_in_54_with_outgroup.txt 4 /mnt/2A2A43C42A438C2F/lab_data/busco_phylo/2021-11-29/orthodb_info_buscos_complete_in_54_with_outgroup.tsv
# uploaded the info file to the server just as a backup
scp -P 2772 orthodb_info_buscos_complete_in_54_with_outgroup.tsv maycon@150.164.28.21:/home/maycon/eusociality/work/busco_phylo/2021-11-29/

# to check the info in a nice, ordered way
tail -n +2 orthodb_info_buscos_complete_in_54_with_outgroup.tsv | LC_ALL=C sort -t "    " -g -r -k3,3 | less -S
```
- Preliminarily selected the following orthologs for the philogeny, as they are the Holometabola equivalent to the original 33 used in the previous tree:
	- 45168at33392;
	- 75072at33392 - trimmed the 3' end (there was a non-coding gap there in multiple sequences);
	- 67937at33392;
	- 65545at33392;
	- 73136at33392;
	- 87096at33392;
	- 75566at33392;
	- 89570at33392;
	- 118220at33392;
	- 116590at33392;
	- 37660at33392;
	- 105333at33392;
	- 87633at33392;
	- 68602at33392;
	- 61549at33392;
	- 74309at33392;
	- 87827at33392;
	- 95275at33392;
	- 24031at33392;
	- 112028at33392;
	- 61826at33392;
	- 79473at33392;
	- 47016at33392 - trimmed 5' and 3' ends;
	- 81580at33392 - trimmed 3' end;
	- 33501at33392 - trimmed a large portion of the 5' end;
	- 59339at33392 - trimmed a large portion of the 5' end;
	- 56574at33392 - trimmed 5' end and a large portion of the 3' end;
	- 86363at33392 - trimmed 5' and 3' ends;
	- 61724at33392 - trimmed 5' and 3' ends, there's a large indel at 3' in two species;
	- 100935at33392 - trimmed 5' and 3' ends;
	- 60560at33392 - trimmed 5' and 3' ends;
	- 114758at33392 - trimmed the 5' end;
	- 116109at33392 - trimmed 5' and 3' ends, alignment at 5' is not great and have many gaps, but they're all multiple of 3 and translation is fine;
	- 56271at33392 - trimmed 5' and 3' ends;
	- 123921at33392 - trimmed the 5' end;
	- 100122at33392 - trimmed the 5' end;
	- 107482at33392 - trimmed 5' and 3' ends;
	- 14326at33392 - trimmed the 5' end;
	- 122669at33392 - trimmed the 5' end;
	- 43062at33392 - trimmed a large portion of the 5' end, and trimmed the 3' end;
	- 43658at33392 - trimmed a large portion of the 5' end, and trimmed the 3' end;
	- 124625at33392 - trimmed 5' and 3' ends;
	- 122095at33392 - trimmed the 3' end.