# 2021-04-16

- All of the following steps were executed inside LGM:
    - Second `MAKER2` run took 31h. The following steps are used after the secon `MAKER2` run to train SNAP a second time;
        - Created a `snap2` folder to store the files;
        - Ran `/usr/local/src/maker/src/bin/gff3_merge -d ../GCF_003254395.2_Amel_HAv3.1_genomic_master_datastore_index.log -o amel_genome.all.gff` to generate the GFF3 input files;
        - Ran `/usr/local/src/maker/src/bin/maker2zff amel_genome.all.gff`  to generate the `genome.ann` and `genome.dna` files;
        - Ran `fathom genome.ann genome.dna -validate > snap_validate_output.txt` to validate the gene models. Checked the validation results with `cat snap_validate_output.txt | grep "error"`. There were no erroneous models;
        - Generated the SNAP input files with the following sequence of commands:
            - `fathom genome.ann genome.dna -categorize 1000`;
            - `fathom uni.ann uni.dna -export 1000 -plus`;
            - `forge export.ann export.dna`;
        - Trained SNAP with `hmm-assembler.pl apis_mellifera . > apis_mellifera.hmm`;
        - Altered the config file to train SNAP again, changing the following options: `maker_gff` and `snaphmm` to point to the `snap2` folder;
        - Ran maker again with `nice -n 10 time maker`;
- Rewritten some of the gene finding script to produce outputs by gene instead of by species and to accept as input a more structured config file;
- Ran the gene finder locally. It worked. Took 14 minutes. Now I have to implement multithreading (if possible) and benchmark it;
- Updated the `git` repo accordingly.