# 2021-05-04

- Did a test run with the entire mitogenome pipeline using a different *Apis mellifera* SRA dataset, with mapQ quality thresshold = 30;
	- Results were terrible as well;
- Found out the issue with the CAFE5 input files and fixed it (`dos2unix`). Unfortunately, CAFE5 is memory intensive and I can't run it locally, so I'm waiting for Chico's clearance to put it on LGM and run it there;
- Installed CAFE5 in LAB;
	- Yep, it ate the entire RAM;
- Tried the mitogenome pipeline with data from another species: *Formica exsecta*. Since it has a smaller volume of data, might as well try it without pre-filtering with BWA and samtools;
	- It yielded 0 reads. Fixed some issues with the script and ran again with a lower mapQ thresshold (20);
		- It still yielded 0 reads;
	- Tried to run MitoZ using the raw FASTQ data;
- Updated the SRA runs for all available species based on the specifications from the MitoZ documentation;