# 2021-10-13

- Found a beetle to use as the outgroup for our phylogeny, [*Tribolium castaneum*](https://www.ncbi.nlm.nih.gov/genome/?term=Tribolium+castaneum%5Borgn%5D). Chose this beetle because:
	- [Peters et al. 2017](https://www.sciencedirect.com/science/article/pii/S0960982217300593) uses it as an outgroup to Hymenoptera (Apocrita + Symphyta, which is our case);
	- [TimeTree](http://timetree.org/) puts Hymenoptera as a sister group of a large clade that includes Coleoptera;
		![[Neoptera_order.jpg]]
	- Genome is annotated on Genbank;
- Downloaded the beetle's genome;
- Did all the standard filtering and preparing procedures to the genome:
```bash
# Extracted orfs
perl ~/eusociality/bin/komodize_genome/bin/extract_ORF.pl genomes/ orfs/

# Got the longest seqs and protein ids for it
perl ~/lab_utils/get_longest_seq.pl orfs/GCF_000002335.3_Tcas5.2_genomic.gbff_nt.fasta > longest/GCF_000002335.3_Tcas5.2_genomic.gbff_nt.fasta.longest

# Extracted the protein ids
grep ">" longest/GCF_000002335.3_Tcas5.2_genomic.gbff_nt.fasta.longest | awk -F"protein_id:" '{print $2}' | awk -F"|" '{print $1}' > prot_ids/GCF_000002335.3_Tcas5.2_genomic.gbff_nt.fasta.longest.ids

# Downloaded the proteome
cd proteins
perl ~/lab_utils/get_seq_efetch.pl ../prot_ids/GCF_000002335.3_Tcas5.2_genomic.gbff_nt.fasta.longest.ids
```
- Ran BUSCO for all the Hymenoptera species + the beetle to get the orthologs that are conserved in all of them, using the Endopterygota dataset `endopterygota_odb10`:
```bash
conda activate BUSCO5
cd /home/maycon/eusociality/results/busco_results/2021-10-13

# Create a folder to store all proteomes
mkdir tmp
# All proteomes are in this directory which I used to get the sequences from the orthodb ID
cp ~/eusociality/work/busco_phylo/2021-09-30/tmp/prot/* tmp/
cp ~/eusociality/data/outgroup/2021-10-13/proteins/GCF_000002335.3_Tcas5.2_genomic.gbff_nt.fasta.longest.ids.aa.fa tmp/

nice -n 10 busco -i tmp/ -o endopterygota_52hymenoptera_1coleoptera_busco -m prot -l endopterygota_odb10 -c 50
```
- Updated the metadata table with the new information and the BUSCO results;
- Started working on getting the BUSCO orthologs for the tree construction:
```bash
# first, move the directories with the raw sequences to recover later to the new directory
cd /home/maycon/eusociality/work/busco_phylo/2021-10-13
mv ../2021-09-30/tmp/ .
# copy the genome that's missing
cp ~/eusociality/data/outgroup/2021-10-13/longest/GCF_000002335.3_Tcas5.2_genomic.gbff_nt.fasta.longest tmp/cds/
cp ~/eusociality/data/outgroup/2021-10-13/proteins/GCF_000002335.3_Tcas5.2_genomic.gbff_nt.fasta.longest.ids.aa.fa tmp/prot/

# get the ids for the BUSCO orthologs classified as "complete"
for file in $(find ~/eusociality/results/busco_results/2021-10-13/endopterygota_52hymenoptera_1coleoptera_busco/*/run*/ -name "full_table*.tsv"); do grep -v "^#" ${file} | awk '$2=="Complete" {print $1}' >> complete_busco_ids.txt; done # 52 Hymenoptera + 1 Coleoptera -> Endopterygota dataset

# get the counts for each unique BUSCO id
sort complete_busco_ids.txt | uniq -c > complete_busco_ids_with_counts.txt

# filter all ids with a count < 53 
# means we only keep those which are complete in all species, Hymenoptera + Coleoptera
awk '$1 > 52 && $1 < 54 {print $2}' complete_busco_ids_with_counts.txt > final_busco_ids_complete_in_53_with_outgroup.txt

# generate the input file for the python script
# getting the busco id, protein id and genome file name
for file in $(find ~/eusociality/results/busco_results/2021-10-13/endopterygota_52hymenoptera_1coleoptera_busco/*/run* -name "full_table*.tsv"); do short=$(echo "${file#*endopterygota_52hymenoptera_1coleoptera_busco/}"); echo "${short%%\.*}" >> protein_ids_complete_in_53_with_outgroup.txt; for id in $(cat final_busco_ids_complete_in_53_with_outgroup.txt); do grep -v "^#" ${file} | awk -v id="$id" '{OFS="|"} {if ($1==id) print $1,$3}' >> protein_ids_complete_in_53_with_outgroup.txt; done; done # all 53 species, including outgroup

# ran the python script to generate separate files for each ortholog
# complete in all species, containing the sequences for all 53 species
# for proteins and CDSs
mkdir genes
python3.6 ~/eusociality/bin/utils/get_busco_genes.py protein_ids_complete_in_53_with_outgroup.txt tmp/cds/ tmp/prot/ genes/

# checked that all generated files have 53 sequences each
grep -c ">" genes/* | awk -F":" '{OFS=":"} {if ($2 != 53) print $1,$2}'
```
- Ran alignments for all CDSs using MAFFT with the `--auto` option (add this to the paper methodology):
```bash
mkdir alignments
nice -n 10 time ~/eusociality/bin/utils/./run_alignments.sh genes/ alignments/ 50
```
- Updated the names to id file to add the new species;
- Ran the script to fetch the OrthoDB data for the BUSCO orthologs to download the data for the 534 genes complete in all 53 species:
```bash
# in /mnt/2A2A43C42A438C2F/lab_data/busco_genes/2021-10-13
scp -P 2772 maycon@150.164.28.21:/home/maycon/eusociality/work/busco_phylo/2021-10-13/final_busco_ids_complete_in_53_with_outgroup.txt .

# in /home/maycon/Documents/LAB/eusociality/local_work
python get_ortodb_data.py /mnt/2A2A43C42A438C2F/lab_data/busco_genes/2021-10-13/final_busco_ids_complete_in_53_with_outgroup.txt 4 /mnt/2A2A43C42A438C2F/lab_data/busco_genes/2021-10-13/orthodb_info_buscos_complete_in_53_with_outgroup.tsv
# uploaded the info file to the server just as a backup
scp -P 2772 orthodb_info_buscos_complete_in_53_with_outgroup.tsv maycon@150.164.28.21:/home/maycon/eusociality/work/busco_phylo/2021-10-13/

# to check the info in a nice, ordered way
tail -n +2 orthodb_info_buscos_complete_in_53_with_outgroup.tsv | LC_ALL=C sort -t "    " -g -r -k3,3 | less -S
```
- Preliminarily selected the following orthologs for the philogeny, as they are the Holometabola equivalent to the original 33 used in the previous tree:
	- 45168at33392;
	- 75072at33392;
	- 67937at33392;
	- 65545at33392;
	- 73136at33392;
	- 87096at33392;
	- 75566at33392;
	- 89570at33392;
	- 118220at33392;
	- 116590at33392;
	- 37660at33392;
	- 105333at33392;
	- 87633at33392;
	- 68602at33392;
	- 61549at33392;
	- 74309at33392;
	- 87827at33392;
	- 95275at33392;
	- 24031at33392;
	- 112028at33392;
	- 61826at33392;
	- 79473at33392;
	- 47016at33392 - trimmed 5' and 3' ends;
	- 81580at33392 - trimmed 3' end;
	- 33501at33392 - trimmed a large portion of the 5' end;
	- 59339at33392 - trimmed a large portion of the 5' end;
	- 56574at33392 - trimmed 5' end and a large portion of the 3' end;
	- 86363at33392 - trimmed 5' and 3' ends;
	- 61724at33392 - trimmed 5' and 3' ends, there's a large indel at 3' in two species;
	- 100935at33392 - trimmed 5' and 3' ends;
	- 60560at33392 - trimmed 5' and 3' ends;
	- 114758at33392 - trimmed the 5' end;
	- 116109at33392 - trimmed 5' and 3' ends, alignment at 5' is not great and have many gaps, but they're all multiple of 3 and translation is fine.