# 2021-04-07

- Worked on the mitogenome downloader script a bit more, but couldn't come up with a solution for all edge cases so gave up on it for now;
- Downloaded all the mitochondrial genomes manually and uploaded them to LGM;
- Install `sra-toolkit` locally;
- Downloaded the **[SRA data for *Bombus impatiens*](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR1575230)** for the same BioSample of the genome we're using, locally;
- Used `prefetch` to download the data and `fastq-dump` to convert the SRA file into 2 FASTQ files (with `--split-files`, being one for the forward and one for the reverse read). The FASTQ files are saved in the HD rather than the SSD since they're quite large and wouldn't fit in the SSD;
- Downloaded [**norgal](https://bitbucket.org/kosaidtu/norgal/src/master/)** and ran it locally with the *Bombus impatiens* FASTQ files and the FASTA genome file;
    - Ran with: `python norgal.py -i /mnt/2A2A43C42A438C2F/lab_data/sra-data/fastq/2021-07-04/SRR1575230_1.fastq /mnt/2A2A43C42A438C2F/lab_data/sra-data/fastq/2021-07-04/SRR1575230_2.fastq -o ~/Documents/LAB/eusociality/local_data/mitogenomes/assembled/2021-07-04 -r ~/Documents/LAB/eusociality/local_data/genomes/2021-07-04/GCF_000188095.2_BIMP_2.1_genomic.fna --trim --delete`;
    - Files were getting too large so ran again specifying a new ouput folder in the HD drive: `o /mnt/2A2A43C42A438C2F/lab_data/mitogenomes/assembled/2021-07-04`.