# 2021-04-14

- All of the following steps were executed inside LGM:
    - First `MAKER2` run took 68h. The following steps are used after the first `MAKER2` run to train the predictors:
        - ~~Created a GFF3 file with the preliminary predictions for the first `MAKER2` run for *Apis mellifera* with: `/usr/local/src/maker/src/bin/gff3_merge GCF_003254395.2_Amel_HAv3.1_genomic_master_datastore_index.log`. This creates the file `genome.all.gff`, which will be used to train `SNAP`;~~
            - ~~This would fail with subsequent commands (due to filtering or something), so ran again with: `/usr/local/src/maker/src/bin/gff3_merge -d GCF_003254395.2_Amel_HAv3.1_genomic_master_datastore_index.log -o amel_genome.all.gff -g`, where `-g` makes it only write `MAKER` gene models to the GFF file and ignore other evidence;~~
            - ~~In addition, used: `/usr/local/src/maker/src/bin/fasta_merge -d GCF_003254395.2_Amel_HAv3.1_genomic_master_datastore_index.log` to generate FASTA files for the protein and transcript evidences (no clue where to use those files);~~
            - ~~Used `/usr/local/src/maker/src/bin/maker2zff amel_genome.all.gff` to generate the input files for SNAP;~~
            - ~~Ran `fathom genome.ann genome.dna -validate > snap_validate_output.txt` to validate the genome files for prediction errors, then used `grep -vwE "MODEL7720" genome.ann > genome.ann2` to remove the erroneous models. Removed the old `genome.ann` file and renamed the new one to `genome.ann`;~~
        - Deleted the previous results and created a new `snap` folder to organize better. Ran everything again:
            - This time, doing `/usr/local/src/maker/src/bin/gff3_merge -d ../GCF_003254395.2_Amel_HAv3.1_genomic_master_datastore_index.log -o amel_genome.all.gff` seem to have done the trick, and the generated gff file seems fine - no errors were printed;
            - Ran `/usr/local/src/maker/src/bin/maker2zff amel_genome.all.gff`, which generated non-empty `genome.ann` and `genome.dna` files (finally!);
            - Ran `fathom genome.ann genome.dna -validate > snap_validate_output.txt` to validate the gene models. Removed the sole erroneous model with `grep -vwE "MODEL7720" genome.ann > genome.ann2`, deleted the old `genome.ann` file and renamed the new one to `genome.ann`. Confirmed the validation with `fathom genome.ann genome.dna -validate`, which outputted no errors;
            - Generated the SNAP input files with the following sequence of commands:
                - `fathom genome.ann genome.dna -categorize 1000`;
                - `fathom uni.ann uni.dna -export 1000 -plus`;
                - `forge export.ann export.dna`;
            - Trained SNAP with `hmm-assembler.pl apis_mellifera . > apis_mellifera.hmm`;
            - Renamed the first run MAKER options file to `old_maker_opts.backup` (to not lose the track of what we've done) and created a `new maker_opts.ctl` file for the second run. In this file, removed the paths in the options `est`, `altest` and `protein`, and switched the `est2genome` and `protein2genome` options to `0`. Added the path to the SNAP hmm file with `snaphmm`;
            - Ran maker again with `nice -n 10 time maker -c 50`. Took 1h24m to run.
            - Created a folder `snap2` to hold the files for the second round of training;
                - ~~Reran all the training steps above within the new folder, for the new result files after the first training round;~~
                - Something went wrong with the first trained run so made some changes to the config file and ran again:
                    - Left back again the path to the `est`, `proteins` and `altest` files;
                    - Added the path to the maker GFF3 file from the first run (`snap/amel_genome.all.gff`) to the `maker_gff` option;
                    - Added the option to use 50 cores for BLAST to the config file;
                    - Ran MAKER once again with `nice -n 10 time maker`;
- Locally:
    - Downloaded all genomes we're using (the old ones and the new annotated and non-annotated ones);
    - Read about the [**rRNA genes**](https://en.wikipedia.org/wiki/Ribosomal_DNA);
    - Started working on a script to retrieve a set of genes from a set of GBFF files.