# 2021-08-03

- Abandomned the genes curation idea;
- Started working on a BUSCO ortholog-based approach, adapting what's described [in this tutorial](https://bioinformaticsworkbook.org/phylogenetics/reconstructing-species-phylogenetic-tree-with-busco-genes-using-maximum-liklihood-method.html#gsc.tab=0);
	- Filtered for "Complete" genes present in all 45 genomes;
- Created a [couple of scripts](https://github.com/flayner2/busco-phlogeny-pipeline) to serve as a general pipeline for the group;
- Finished the initial filtering the BUSCO data, creating fasta files with all CDS and all proteins for the alignment;
- Compiled [Guidance](http://guidance.tau.ac.il/source.php) in my personal `bin` folder in LGM;
	- Need to install Ruby;
- Created a script to run the alignments for all genes.