# 2021-04-20

- All of the following steps were executed inside LGM:
    - Continuing with the AUGUSTUS training:
        - The optimization step took about 2 days to complete;
        - Generated the second training results with:
            - `etraining --species=apis_mellifera augustus.gbk.train`;
            - `augustus --species=apis_mellifera augustus.gbk.test | tee second_training.out`;
    - The results for before and after the optimization step are (checked with `grep -A 22 Evaluation FILENAME.out`, where `FILENAME` is the name of the first or the second training output file):
        - First training:
            - nucleotide level: 0.968 sensitivity, 0.853 specificity;
            - exon level: 0.83 sensitivity, 0.768 specificity;
            - gene level: 0.46 sensitivity, 0.351 specificity;
        - Second training:
            - nucleotide level: 0.97 sensitivity, 0.857 specificity;
            - exon level: 0.834 sensitivity, 0.775 specificity;
            - gene level: 0.46 sensitivity, 0.354 specificity;
    - Changed the MAKER options: `augustus_species=apis_mellifera` and `keep_preds=1`. Also had to fix the `AUGUSTUS_CONFIG_PATH` environmental variable by providing the full path to the local `config` directory;
    - Ran maker again with `nice -n 10 time maker`;
    - Re-downloaded the reference mitogenomes as FASTA files and uploaded to LGM;
    - Ran ~~KOMODO2~~ CALANGO on old Pfam data to generate a matrix of domain counts for each Hymenoptera species used in the previous analysis. Sent that file and the ultrametric tree used for my dissertation to `labbioinfo`;
    - Fixed the naming of the species in the matrix file and the tree;
    - Ran `CAFE` on `labbioinfo`.