# 2021-09-30

- Generated the tabular files from the InterProScan output:
```bash
# for GO
perl ~/eusociality/bin/KOMODO2/bin/utils/parse_interproscan_output_gene2GO.pl GCF_014066325.tsv /home/maycon/eusociality/work/komodo_inputs/gene2go/2021-09-29/

# for Pfam
perl ~/eusociality/bin/KOMODO2/bin/utils/parse_interproscan_output.pl GCF_014066325.tsv "Pfam" /home/maycon/eusociality/work/komodo_inputs/pfam/2021-09-29/
```
- Renamed the files to abide to the names in the tree nodes, using the short names defined in the metadata sheet (first updated `names.txt` to contain all the new names):
```bash
# from inside the tabular data directory
~/eusociality/bin/utils/./create_renamed_symlinks.sh ~/eusociality/bin/utils/names.txt .
```
- Counted the features in the tabular files:
```bash
# for GO
for file in *.txt; do perl ~/eusociality/bin/KOMODO2/bin/utils/count_GO_gene2GO.pl $file >> counts.txt; done

# for Pfam
for file in *.txt; do printf "$file" >> counts.txt && tail -n +2 $file | wc -l | awk 'OFS="\t" {print $2, $1}' >> counts.txt; done

# to rename
~/eusociality/bin/utils/./rename_counts.sh ~/eusociality/bin/utils/names.txt counts.txt
```
- Backed up and updated the metadata files in `/home/maycon/eusociality/data/metadata/masters_paper_metadata` with the data for the new species;
- Added the metadata for *Apis laboriosa* to the metadata file:
```bash
# formatted the raw data file
awk -F'\t' 'OFS="\t" {print $2, $5, 1 + (log($5)/log(10.0)), $6, $4, $1, $3}' apis_laboriosa_data.txt > apis_laboriosa_formatted.txt

# pasted the counts with the formatted data files
paste apis_laboriosa_counts.go apis_laboriosa_formatted.txt > apis_laboriosa_go.tsv
paste apis_laboriosa_counts.pfam apis_laboriosa_formatted.txt > apis_laboriosa_pfam.tsv

# edited the files to remove uncessary columns and then...
cat metadata_gene2go_51hymenoptera_final.tsv apis_laboriosa_go.tsv | sort -k1,1 > metadata_gene2go_52hymenoptera_final.tsv
cat metadata_pfam_51hymenoptera_final.tsv apis_laboriosa_pfam.tsv | sort -k1,1 > metadata_pfam_52hymenoptera_final.tsv
```
- Started working on getting the BUSCO orthologs for the tree construction:
```bash
# get the ids for the BUSCO orthologs classified as "complete"
cd /home/maycon/eusociality/work/busco_phylo/2021-09-30
cp ../2021-09-24/complete_busco_ids.txt .
grep -v "^#" ~/eusociality/results/busco_results/2021-09-29/apis_laboriosa_busco/run_hymenoptera_odb10/full_table.tsv | awk '$2=="Complete" {print $1}' >> complete_busco_ids.txt # Apis laboriosa

# get the counts for each unique BUSCO id
sort complete_busco_ids.txt | uniq -c > complete_busco_ids_with_counts.txt

# filter all ids with a count < 52 
# means we only keep those which are complete in all species
awk '$1 > 51 && $1 < 53 {print $2}' complete_busco_ids_with_counts.txt > final_busco_ids_complete_in_52.txt

# generate the input file for the python script
# getting the busco id, protein id and genome file name
for file in $(find ~/eusociality/results/busco_results/2021-08-10/hymenoptera_all_annotated/*/run* -name "full_table*.tsv"); do short=$(echo "${file#*hymenoptera_all_annotated/}"); echo "${short%%\.*}" >> protein_ids_complete_in_52.txt; for id in $(cat final_busco_ids_complete_in_52.txt); do grep -v "^#" ${file} | awk -v id="$id" '{OFS="|"} {if ($1==id) print $1,$3}' >> protein_ids_complete_in_52.txt; done; done # original 44
for file in $(find ~/eusociality/results/busco_results/2021-09-20/hymenoptera_ncbi_annotated_01/*/run* -name "full_table*.tsv"); do short=$(echo "${file#*hymenoptera_ncbi_annotated_01/}"); echo "${short%%\.*}" >> protein_ids_complete_in_52.txt; for id in $(cat final_busco_ids_complete_in_52.txt); do grep -v "^#" ${file} | awk -v id="$id" '{OFS="|"} {if ($1==id) print $1,$3}' >> protein_ids_complete_in_52.txt; done; done # new 5
echo GCF_014825855 >> protein_ids_complete_in_52.txt && for id in $(cat final_busco_ids_complete_in_52.txt); do grep -v "^#" ~/eusociality/results/busco_results/2021-09-23/bombus_pyrosoma_busco/run_hymenoptera_odb10/full_table.tsv | awk -v id="$id" '{OFS="|"} {if ($1==id) print $1,$3}' >> protein_ids_complete_in_52.txt; done # Bombus pyrosoma
echo GCF_014466175 >> protein_ids_complete_in_52.txt && for id in $(cat final_busco_ids_complete_in_52.txt); do grep -v "^#" ~/eusociality/results/busco_results/2021-09-24/vespula_pensylvanica_busco/run_hymenoptera_odb10/full_table.tsv | awk -v id="$id" '{OFS="|"} {if ($1==id) print $1,$3}' >> protein_ids_complete_in_52.txt; done # Vespula pensylvanica
echo GCF_014066325 >> protein_ids_complete_in_52.txt && for id in $(cat final_busco_ids_complete_in_52.txt); do grep -v "^#" ~/eusociality/results/busco_results/2021-09-29/apis_laboriosa_busco/run_hymenoptera_odb10/full_table.tsv | awk -v id="$id" '{OFS="|"} {if ($1==id) print $1,$3}' >> protein_ids_complete_in_52.txt; done # Apis laboriosa

# copied the previously existant temp directory
cp -r ../2021-09-24/tmp/ .

# FOR THE LONGEST CDSs
cp ~/eusociality/data/hymenoptera/2021-09-29/longest/GCF_014066325.1_ASM1406632v1_genomic.gbff_nt.fasta.longest tmp/cds/

# FOR THE PROTEINS
cp ~/eusociality/data/hymenoptera/2021-09-29/proteins/GCF_014066325.1_ASM1406632v1_genomic.gbff_nt.fasta.longest.ids.aa.fa tmp/prot/

# deleted the old tmp folder to save space
rm -rf ../2021-09-24/tmp/

# ran the python script to generate separate files for each ortholog
# complete in all species, containing the sequences for all 52 species
# for proteins and CDSs
mkdir genes
python3.6 ~/eusociality/bin/utils/get_busco_genes.py protein_ids_complete_in_52.txt tmp/cds/ tmp/prot/ genes/

# checked that all generated files have 52 sequences each
grep -c ">" genes/* | awk -F":" '{OFS=":"} {if ($2 != 52) print $1,$2}'
```
- Ran alignments for all CDSs using MAFFT with the `--auto` option (add this to the paper methodology):
```bash
mkdir alignments
nice -n 10 time ~/eusociality/bin/utils/./run_alignments.sh genes/ alignments/ 50
```
- Updated the script to fetch the OrthoDB data for the BUSCO orthologs and ran it locally to download the data for the 1017 genes complete in all 52 species:
```bash
# in /mnt/2A2A43C42A438C2F/lab_data/busco_genes/2021-09-30
scp -rP 2772 maycon@150.164.28.21:/home/maycon/eusociality/work/busco_phylo/alignments .
scp -P 2772 maycon@150.164.28.21:/home/maycon/eusociality/work/busco_phylo/2021-09-30/final_busco_ids_complete_in_52.txt .

# in /home/maycon/Documents/LAB/eusociality/local_work
python get_ortodb_data.py /mnt/2A2A43C42A438C2F/lab_data/busco_genes/2021-09-30/final_busco_ids_complete_in_52.txt 4 $STORAGE/lab_data/busco_genes/2021-09-30/orthodb_info_buscos_complete_in_52.tsv
# uploaded the info file to the server just as a backup
scp -P 2772 orthodb_info_buscos_complete_in_52.tsv maycon@150.164.28.21:/home/maycon/eusociality/work/busco_phylo/2021-09-30/

# to check the info in a nice, ordered way
tail -n +2 orthodb_info_buscos_complete_in_52.tsv | LC_ALL=C sort -t "    " -g -r -k3,3 | less -S
```
- Checked and trimmed the alignments for the following 23 genes:
	- 10155at7399;
	- 10627at7399;
	- 11582at7399;
	- 12100at7399;
	- 15152at7399;
	- ~~15909at7399~~: **not present in all species**;
	- 16487at7399: **cut the non-coding gap at 5'**;
	- 17089at7399;
	- 17317at7399;
	- 19260at7399;
	- 21170at7399;
	- 21378at7399;
	- 21938at7399;
	- 22853at7399;
	- 23430at7399;
	- 24551at7399;
	- 25185at7399: **cut the 5' end and transformed a non-coding 5' set of gaps into missing data by replacing them for "?"s**;
	- 26386at7399;
	- 26463at7399;
	- 28301at7399;
	- 28363at7399;
	- 28474at7399;
	- 9827at7399;
- For now, ignored the following 10 genes due to potential issues:
	- 22903at7399;
	- 27781at7399;
	- 10687at7399;
	- 15447at7399;
	- 22204at7399;
	- 18731at7399;
	- 26814at7399;
	- 15773at7399;
	- 20450at7399;
	- 25129at7399;
- Sent both the raw and corrected alignments to ZÃ© for review;
- Added the following genes:
	- 24786at7399;
	- 16855at7399 **tranformed 1 "-" into missing data "?" at the 5' end**;
	- 14667at7399;
	- 7735at7399;
	- 26746at7399;
	- 11316at7399.
