# 2022-01-05

- Started parsing the Orthofinder + Signatures outputs to create the charts:
```bash
grep -f Orthogroups_SingleCopyOrthologues.txt Orthogroups.tsv > single_copy_orthogroups_to_ids.tsv

# Manually edited the output file to add a header column with
# orthogroup    Bimp    Aros    Emex    Amel

# Filtered the signatures odds ratios results
cd /home/maycon/eusociality/results/odds_ratios_genewise
# 'genomes.txt' contains a wildcard-completed name for the 4 genomes
grep -f genomes.txt final_results_dinucleotide_odds_ratio.txt > final_results_dinucleotide_odds_ratio_4species.txt
grep -f genomes.txt odds_ratio_CG_only.txt > odds_ratio_CG_only_4species.txt
# 'subs.txt' contains simple substitution patterns
sed -f subs.txt odds_ratio_CG_only_4species.txt > odds_ratio_CG_only_4species_renamed.txt
sed -f subs.txt final_results_dinucleotide_odds_ratio_4species.txt > final_results_dinucleotide_odds_ratio_4species_renamed.txt

# Downloaded the files locally and proceeded with parsing and filtering
# This gets all single-copy orthologs gene ids, used to filter the singatures files
for i in {2..5}; do tail -n +2 single_copy_orthogroups_to_ids.tsv | awk -v col="$i" '{print $col}' >> single_copy_geneids.txt; done
grep -f single_copy_geneids.txt odds_ratio_CG_only_4species_renamed.txt > odds_ratio_CG_only_4species_renamed_single_copy_only.txt
grep -f single_copy_geneids.txt final_results_dinucleotide_odds_ratio_4species_renamed.txt  > final_results_dinucleotide_odds_ratio_4species_renamed_single_copy_only.txt
```
- Started working on the R script to perform the analyses and generate the visualizations;
- Generated some initial scatter plots and shared them with Chico.