# 2022-01-18

## Paper
- Pesquisar tudo sobre biologia de *Athalia* pra justificar o padrão bimodal de CpG;
- ~~Conferir se os hipometilados são os mesmos mesmo numa distância filogenética grande (paper do PNAS e do GBE)~~;
- Focar na ideia do teste que fizemos do fold-change (take home message);
	- Bater **muito** na tecla da via TOR;
- Levantar experimentos de transcriptoma pra castas de *Apis*;
- Fazer um [paired boxplot](https://rpkgs.datanovia.com/ggpubr/reference/ggpaired.html)
- Revisão sucinta, maximo de 15 referências (no total) - falar do estado da arte e do problema que a gente investigou;
	- Metilação é importante pra diferenciação de castas;
	- Até hoje não sabemos o mecanismo epigenético direito;
	- O paper do PNAS vende uma ideia, mas o da GBE e nossos dados contrastam;
- ~~Olhar as citações dos papers da PNAS e da GBE~~;
- Status das DNMTs nos genomas desses bichos;
- Lá no final, pra _confirmação_, a gente fala sobre as outras abelhas e a *Harpegnathos* e *Solenopsis* e *Cephus* e *Trichogramma* e *Polistes canadensis*, as utilmas unimodais com presença de DNMT3 (calcular os ortologos 1:1 incluindo eles e fazer os scatter com *Apis* no eixo X);
- Mirar como se fosse um Brief Report da PNAS.

## Work
- Ran Orthofinder including *H. saltator*, *S. invicta*, *C. cinctus*, *T. pretiosum* e *P. canadensis*:
```bash
# Getting the inputs to one place
cd orthofinder/2022-01-18
cp -r ../2022-01-12/inputs/ .
cd inputs
cp ~/eusociality/data/hymenoptera/2019-11-07/proteins/hymenoptera_proteins_34.tar.gz .
tar -xvzf hymenoptera_proteins_34.tar.gz GCF_000341935.1_Ccin1_genomic.gbff_nt.fasta.longest.fasta.ids.txt.aa.fa GCF_003227715.1_Hsal_v8.5_genomic.gbff_nt.fasta.longest.fasta.ids.txt.aa.fa GCF_001313835.1_ASM131383v1_genomic.gbff_nt.fasta.longest.fasta.ids.txt.aa.fa GCF_000188075.2_Si_gnH_genomic.gbff_nt.fasta.longest.fasta.ids.txt.aa.fa GCF_000599845.2_Tpre_2.0_genomic.gbff_nt.fasta.longest.fasta.ids.txt.aa.fa
rm hymenoptera_proteins_34.tar.gz
cd ..

# Running Othofinder
screen -S orthofinder
conda activate orthofinder
nice -n 10 orthofinder -t 55 -f inputs/ -n cpg_hymenoptera_10species_with_unimodal_and_dmel
mv inputs/OrthoFinder/Results_cpg_hymenoptera_10species_with_unimodal_and_dmel/ ../../../results/orthofinder_results/2022-01-18/
grep -f Orthogroups_SingleCopyOrthologues.txt Orthogroups.tsv > single_copy_orthogroups_to_ids.tsv
```
- Downloaded and processed the results:
```bash
# Downloaded the files locally
# Concatenated and renamed entries in the odds ratios files
for i in `ls to_download/`; do tail -n +2 to_download/$i >> final_results_dinucleotide_odds_ratio_10species_renamed_unimodal_and_dmel.txt; done
cat final_results_dinucleotide_odds_ratio_4species_renamed_inc_dmel.txt final_results_dinucleotide_odds_ratio_10species_renamed_unimodal_and_dmel.txt > final_results_dinucleotide_odds_ratio_10species_renamed_unimodal_and_dmel.txt.tmp
rm final_results_dinucleotide_odds_ratio_10species_renamed_unimodal_and_dmel.txt
mv final_results_dinucleotide_odds_ratio_10species_renamed_unimodal_and_dmel.txt.tmp final_results_dinucleotide_odds_ratio_10species_renamed_unimodal_and_dmel.txt

tail -n +2 single_copy_orthogroups_to_ids_10species_unimodal_and_dmel.tsv | awk 'OFS="\n" {print $2, $3, $4, $5, $6, $7, $8, $9, $10, $11}' > single_copy_geneids_10species_unimodal_and_dmel.txt
grep "CG" final_results_dinucleotide_odds_ratio_10species_renamed_unimodal_and_dmel.txt > odds_ratio_CG_only_10species_renamed_uniodal_and_dmel.txt
dos2unix *.txt
dos2unix *.tsv
grep -f single_copy_geneids_10species_unimodal_and_dmel.txt odds_ratio_CG_only_10species_renamed_uniodal_and_dmel.txt > odds_ratio_CG_only_10species_renamed_single_copy_uniodal_and_dmel.txt
grep -f single_copy_geneids_10species_unimodal_and_dmel.txt final_results_dinucleotide_odds_ratio_10species_renamed_unimodal_and_dmel.txt  > final_results_dinucleotide_odds_ratio_10species_renamed_single_copy_unimodal_and_dmel.txt
```
- Generated the plots including all 10 species.